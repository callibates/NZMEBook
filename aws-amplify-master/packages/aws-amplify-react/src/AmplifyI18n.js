/*
 * Copyright 2017-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
 * the License. A copy of the License is located at
 *
 *     http://aws.amazon.com/apache2.0/
 *
 * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
 * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
 * and limitations under the License.
 */

const dict = {
    'fr': {
        'Sign In': "Se connecter",
        'Sign Up': "S'inscrire",
        'Sign Out': "Déconnexion",
        'Forgot Password': "Mot de passe oublié",
        'Username': "Nom d'utilisateur",
        'Password': "Mot de passe",
        'Change Password': "Changer le mot de passe",
        'New Password': "nouveau mot de passe",
        'Email': "Email",
        'Phone Number': "Numéro de téléphone",
        'Confirm a Code': "Confirmer un code",
        'Confirm Sign In': "Confirmer la connexion",
        'Confirm Sign Up': "Confirmer l'inscription",
        'Back to Sign In': "Retour à la connexion",
        'Send Code': "Envoyer le code",
        'Confirm': "Confirmer",
        'Resend a Code': "Renvoyer un code",
        'Submit': "Soumettre",
        'Skip': "Sauter",
        'Verify': "Vérifier",
        'Verify Contact': "Vérifier le contact",
        'Code': "Code",
        'Account recovery requires verified contact information':
"La récupération du compte nécessite des informations de contact vérifiées",

        'User does not exist': "L'utilisateur n'existe pas",
        'User already exists': "L'utilisateur existe déjà",
        'Incorrect username or password': "identifiant ou mot de passe incorrect",
        'Invalid password format': "format de mot de passe invalide",
        'Invalid phone number format':
`Format de numéro de téléphone invalide.
Veuillez utiliser un format de numéro de téléphone du +12345678900`
    },

    'es': {
        'Sign In': "Registrarse",
        'Sign Up': "Regístrate",
        'Sign Out': "Desconectar",
        'Forgot Password': "Se te olvidó tu contraseña",
        'Username': "Nombre de usuario",
        'Password': "Contraseña",
        'Change Password': "Cambia la contraseña",
        'New Password': "Nueva contraseña",
        'Email': "Email",
        'Phone Number': "Número de teléfono",
        'Confirm a Code': "Confirmar un código",
        'Confirm Sign In': "Confirmar inicio de sesión",
        'Confirm Sign Up': "Confirmar Registrarse",
        'Back to Sign In': "Volver a Iniciar sesión",
        'Send Code': "Enviar código",
        'Confirm': "Confirmar",
        'Resend a Code': "Reenviar un código",
        'Submit': "Enviar",
        'Skip': "Omitir",
        'Verify': "Verificar",
        'Verify Contact': "Verificar contacto",
        'Code': "Código",
        'Account recovery requires verified contact information':
"La recuperación de la cuenta requiere información de contacto verificada",

        'User does not exist': "el usuario no existe",
        'User already exists': "El usuario ya existe",
        'Incorrect username or password': "Nombre de usuario o contraseña incorrecta",
        'Invalid password format': "Formato de contraseña inválido",
        'Invalid phone number format':
`Formato de número de teléfono inválido.
Utilice el formato de número de teléfono +12345678900`
    },

    'zh': {
        'Sign In': "登录",
        'Sign Up': "注册",
        'Sign Out': "退出",
        'Forgot Password': "忘记密码",
        'Username': "用户名",
        'Password': "密码",
        'Change Password': "改变密码",
        'New Password': "新密码",
        'Email': "邮箱",
        'Phone Number': "电话",
        'Confirm a Code': "确认码",
        'Confirm Sign In': "确认登录",
        'Confirm Sign Up': "确认注册",
        'Back to Sign In': "回到登录",
        'Send Code': "发送确认码",
        'Confirm': "确认",
        'Resend a Code': "重发确认码",
        'Submit': "提交",
        'Skip': "跳过",
        'Verify': "验证",
        'Verify Contact': "验证联系方式",
        'Code': "确认码",
        'Account recovery requires verified contact information':
"账户恢复需要验证过的联系方式",

        'User does not exist': "用户不存在",
        'User already exists': "用户已经存在",
        'Incorrect username or password': "用户名或密码错误",
        'Invalid password format': "密码格式错误",
        'Invalid phone number format': "电话格式错误，请使用格式 +12345678900"
    }
}

export default dict;
