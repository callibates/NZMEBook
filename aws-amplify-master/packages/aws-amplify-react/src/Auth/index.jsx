/*
 * Copyright 2017-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
 * the License. A copy of the License is located at
 *
 *     http://aws.amazon.com/apache2.0/
 *
 * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
 * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
 * and limitations under the License.
 */

import React, { Component } from 'react';
import PropTypes from 'prop-types';

import Authenticator from './Authenticator';

export { default as Authenticator } from './Authenticator';
export { default as AuthPiece } from './AuthPiece';
export { default as SignIn } from './SignIn';
export { default as ConfirmSignIn } from './ConfirmSignIn';
export { default as SignOut } from './SignOut';
export { default as RequireNewPassword } from './RequireNewPassword';
export { default as SignUp } from './SignUp';
export { default as ConfirmSignUp } from './ConfirmSignUp';
export { default as VerifyContact } from './VerifyContact';
export { default as ForgotPassword } from './ForgotPassword';
export { default as Greetings } from './Greetings';
export { default as FederatedSignIn, FederatedButtons } from './FederatedSignIn';
export { default as TOTPSetup } from './TOTPSetup';

export * from './Provider';

import Greetings from './Greetings';

export function withAuthenticator(Comp, includeGreetings=false, authenticatorComponents = []) {
    return class extends Component {
        constructor(props) {
            super(props);

            this.handleAuthStateChange = this.handleAuthStateChange.bind(this);

            this.state = {
                authState: props.authState || null,
                authData: props.authData || null
            };
        }

        handleAuthStateChange(state, data) {
            this.setState({ authState: state, authData: data });
        }

        render() {
            const { authState, authData } = this.state;
            const signedIn = (authState === 'signedIn');
            if (signedIn) {
                return (
                    <div>
                        {
                            includeGreetings?
                            <Greetings
                                authState={authState}
                                authData={authData}
                                onStateChange={this.handleAuthStateChange}
                            />
                            : null
                        }
                        <Comp
                            {...this.props}
                            authState={authState}
                            authData={authData}
                            onStateChange={this.handleAuthStateChange}
                        />
                    </div>
                )
            }

            return <Authenticator
                {...this.props}
                hideDefault={authenticatorComponents.length > 0}
                onStateChange={this.handleAuthStateChange}
                children={authenticatorComponents}
            />
        }
    }
}

export class AuthenticatorWrapper extends Component {
    constructor(props) {
        super(props);

        this.state = { auth: 'init' };

        this.handleAuthState = this.handleAuthState.bind(this);
    }

    handleAuthState(state, data) {
        this.setState({ auth: state, authData: data });
    }

    render() {
        return (
            <div>
                <Authenticator {...this.props} onStateChange={this.handleAuthState} />
                {this.props.children(this.state.auth)}
            </div>
        )
    }
}

AuthenticatorWrapper.propTypes = {
    children: PropTypes.func.isRequired
}
