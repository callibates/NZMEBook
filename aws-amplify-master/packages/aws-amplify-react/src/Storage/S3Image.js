/*
 * Copyright 2017-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
 * the License. A copy of the License is located at
 *
 *     http://aws.amazon.com/apache2.0/
 *
 * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
 * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
 * and limitations under the License.
 */

import React, { Component } from 'react';

import { Storage, Logger } from 'aws-amplify';

import AmplifyTheme from '../AmplifyTheme';
import { transparent1X1 } from '../AmplifyUI';
import { PhotoPicker } from '../Widget';
import { calcKey } from './Common';

const logger = new Logger('Storage.S3Image');

export default class S3Image extends Component {
    constructor(props) {
        super(props);

        this.handleOnLoad = this.handleOnLoad.bind(this);
        this.handleOnError = this.handleOnError.bind(this);
        this.handlePick = this.handlePick.bind(this);
        this.handleClick = this.handleClick.bind(this);

        const initSrc = this.props.src || transparent1X1;

        this.state = { src: initSrc };
    }

    getImageSource(key, level, track) {
        Storage.get(key, { level: level? level : 'public', track })
            .then(url => {
                this.setState({
                    src: url
                });
            })
            .catch(err => logger.debug(err));
    }

    load() {
        const { imgKey, path, body, contentType, level, track } = this.props;
        if (!imgKey && !path) {
            logger.debug('empty imgKey and path');
            return ;
        }

        const that = this;
        const key = imgKey || path;
        logger.debug('loading ' + key + '...');
        if (body) {
            const type = contentType || 'binary/octet-stream';
            const ret = Storage.put(key, body, {
                contentType: type,
                level: level? level : 'public',
                track
            });
            ret.then(data => {
                logger.debug(data);
                that.getImageSource(key, level, track);
            })
            .catch(err => logger.debug(err));
        } else {
            that.getImageSource(key, level, track);
        }
    }

    handleOnLoad(evt) {
        const { onLoad } = this.props;
        if (onLoad) { onLoad(this.state.src); }
    }

    handleOnError(evt) {
        const { onError } = this.props;
        if (onError) { onError(this.state.src); }
    }

    handlePick(data) {
        const that = this;

        const path = this.props.path || '';
        const { imgKey, level, fileToKey, track } = this.props;
        const { file, name, size, type } = data;
        const key = imgKey || (path + calcKey(data, fileToKey));
        Storage.put(key, file, { 
            level: level? level: 'public',
            contentType: type, 
            track 
        })
            .then(data => {
                logger.debug('handle pick data', data);
                that.getImageSource(key, level, track);
            })
            .catch(err => logger.debug('handle pick error', err));
    }

    handleClick(evt) {
        const { onClick } = this.props;
        if (onClick) { onClick(evt); }
    }

    componentDidMount() {
        this.load();
    }

    componentDidUpdate(prevProps) {
        const update = prevProps.path !== this.props.path ||
                        prevProps.imgKey !== this.props.imgKey ||
                        prevProps.body !== this.props.body;
        if (update) {
            this.load();
        }
    }

    imageEl(src, theme) {
        if (!src) { return null; }

        const { selected } = this.props;
        const containerStyle = { position: 'relative' };
        return (
            <div style={containerStyle} onClick={this.handleClick}>
                <img
                    style={theme.photoImg}
                    src={src}
                    onLoad={this.handleOnLoad}
                    onError={this.handleOnError}
                />
                <div style={selected? theme.overlaySelected : theme.overlay}></div>
            </div>
        )
    }

    render() {
        const { hidden, style, picker, translate, imgKey } = this.props;
        let src = this.state.src;
        if (translate) {
            src = (typeof translate === 'string')? translate : translate({
                type: 'image',
                key: imgKey,
                content: src
            });
        }
        if (!src && !picker) { return null; }

        const theme = this.props.theme || AmplifyTheme;
        const photoStyle = hidden? AmplifyTheme.hidden
                                 : Object.assign({}, theme.photo, style);

        return (
            <div style={photoStyle}>
                { photoStyle? this.imageEl(src, theme): null }
                { picker? <div>
                              <PhotoPicker
                                  key="picker"
                                  onPick={this.handlePick}
                                  theme={theme}
                              />
                          </div> : null
                }
            </div>
        )
    }
}
