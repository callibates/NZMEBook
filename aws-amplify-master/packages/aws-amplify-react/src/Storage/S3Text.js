/*
 * Copyright 2017-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
 * the License. A copy of the License is located at
 *
 *     http://aws.amazon.com/apache2.0/
 *
 * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
 * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
 * and limitations under the License.
 */

import React, { Component } from 'react';

import { Storage, Logger } from 'aws-amplify';

import AmplifyTheme from '../AmplifyTheme';
import { TextPicker } from '../Widget';
import { calcKey } from './Common';

const logger = new Logger('Storage.S3Text');

export default class S3Text extends Component {
    constructor(props) {
        super(props);

        this.handleOnLoad = this.handleOnLoad.bind(this);
        this.handleOnError = this.handleOnError.bind(this);
        this.handlePick = this.handlePick.bind(this);
        this.handleClick = this.handleClick.bind(this);

        const { text, textKey } = props;
        this.state = {
            text: text || '',
            textKey: textKey || ''
        };
    }

    getText(key, level, track) {
        Storage.get(key, { download: true, level: level? level : 'public', track })
            .then(data => {
                logger.debug(data);
                const text = data.Body.toString('utf8');
                this.setState({ text: text });
                this.handleOnLoad(text);
            })
            .catch(err => {
                logger.debug(err);
                this.handleOnError(err);
            });
    }

    load() {
        const { path, textKey, body, contentType, level, track } = this.props;
        if (!textKey && !path) {
            logger.debug('empty textKey and path');
            return ;
        }

        const that = this;
        const key = textKey || path;
        logger.debug('loading ' + key + '...');
        if (body) {
            const type = contentType || 'text/*';
            const ret = Storage.put(key, body, {
                contentType: type,
                level: level? level : 'public',
                track
            });
            ret.then(data => {
                logger.debug(data);
                that.getText(key, level, track);
            })
            .catch(err => logger.debug(err));
        } else {
            that.getText(key, level, track);
        }
    }

    handleOnLoad(text) {
        const { onLoad } = this.props;
        if (onLoad) { onLoad(text); }
    }

    handleOnError(err) {
        const { onError } = this.props;
        if (onError) { onError(err); }
    }

    handlePick(data) {
        const that = this;

        const path = this.props.path || '';
        const { textKey, level, fileToKey, track } = this.props;
        const { file, name, size, type } = data;
        const key = textKey || (path + calcKey(data, fileToKey));
        Storage.put(key, file, {
            level: level? level: 'public',
            contentType: type,
            track
        })
            .then(data => {
                logger.debug('handle pick data', data);
                that.getText(key, level, track);
            })
            .catch(err => logger.debug('handle pick error', err));
    }

    handleClick(evt) {
        const { onClick } = this.props;
        if (onClick) { onClick(evt); }
    }

    componentDidMount() {
        this.load();
    }

    componentDidUpdate(prevProps) {
        const update = prevProps.path !== this.props.path ||
                        prevProps.textKey !== this.props.textKey ||
                        prevProps.body !== this.props.body
        if (update) {
            this.load();
        }
    }

    textEl(text, theme) {
        if (!text) { return null; }

        const { selected } = this.props;
        const containerStyle = { position: 'relative' };
        return (
            <div style={containerStyle} onClick={this.handleClick}>
                <pre style={theme.pre}>{text}</pre>
                <div style={selected? theme.overlaySelected : theme.overlay}></div>
            </div>
        )
    }

    render() {
        const { hidden, style, picker, translate, textKey } = this.props;
        let text = this.state.text;
        if (translate) {
            text = (typeof translate === 'string')? translate : translate({
                type: 'text',
                key: textKey,
                content: text
            });
        }
        if (!text && !picker) { return null; }

        const theme = this.props.theme || AmplifyTheme;
        const textStyle = hidden? AmplifyTheme.hidden
                                : Object.assign({}, theme.text, style);

        return (
            <div style={textStyle}>
                { textStyle? this.textEl(text, theme) : null }
                { picker? <div>
                              <TextPicker
                                  key="picker"
                                  onPick={this.handlePick}
                                  theme={theme}
                              />
                          </div>
                        : null
                }
            </div>
        )
    }
}
