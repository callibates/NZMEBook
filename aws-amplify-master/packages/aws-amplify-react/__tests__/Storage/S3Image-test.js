jest.mock('../../src/Storage/Common', () => {
    const calcKey = () => {
        return '';
    }

    return { calcKey };
});

import S3Image from '../../src/Storage/S3Image';
import { calcKey } from '../../src/Storage/Common';
import { PhotoPicker } from '../../src/Widget'
import { Storage, Logger } from 'aws-amplify';
import React from 'react';

describe('S3Image', () => {
    describe('render test', () => {
        test('render null if no test and no picker', () => {
            const wrapper = shallow(<S3Image/>);

            expect(wrapper).toMatchSnapshot();
        });

        test('render with picker true', () => {
            const wrapper = shallow(<S3Image picker/>);
            
            expect(wrapper).toMatchSnapshot();
        });

        test('render with src exist', () => {
            const wrapper = shallow(<S3Image/>);
            wrapper.setState({src: 'imageSrc'});

            expect(wrapper).toMatchSnapshot();
        });

        test('render with translate text', () => {
            const wrapper = shallow(<S3Image translate='translate'/>);

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe('interaction test', () => {
        test('PhotoPicker onPick test', () => {
            const spyon = jest.spyOn(S3Image.prototype, 'handlePick')
                .mockImplementationOnce(() => {
                    return;
                });

            const wrapper = shallow(<S3Image picker/>);
                
            wrapper.find(PhotoPicker).simulate('pick');

            expect(spyon).toBeCalled();

            spyon.mockClear();
        });
        
        test('imageEl click test', () => {
            const spyon = jest.spyOn(S3Image.prototype, 'handleClick')
                .mockImplementationOnce(() => {
                    return;
                });
            const wrapper = shallow(<S3Image/>);
            wrapper.setState({src: 'imageSrc'});

            wrapper.find('div').at(1).simulate('click');

            expect(spyon).toBeCalled();

            spyon.mockClear();
        });
    });

    describe('load test', () => {
        test('happy case with body set', async () => {
            const spyon = jest.spyOn(Storage, 'put').mockImplementationOnce(() => {
                return new Promise((res, rej) => {
                    res('data');
                });
            });

            const spyon2 = jest.spyOn(S3Image.prototype, 'getImageSource').mockImplementationOnce(() => {
                return;
            });

            const props = {
                imgKey: 'imgKey',
                body: 'imgBody'
            };

            const wrapper = shallow(<S3Image/>);
            const s3Image = wrapper.instance();
            wrapper.setProps(props);

            await s3Image.load();

            expect.assertions(2);
            expect(spyon).toBeCalledWith('imgKey', 'imgBody', { contentType: 'binary/octet-stream', level: 'public' });
            expect(spyon2).toBeCalled();

            spyon.mockClear();
            spyon2.mockClear();
        });

        test('nothing to do if no imgKey and path set', async () => {
            const spyon = jest.spyOn(Logger.prototype, 'debug');

            const wrapper = shallow(<S3Image/>);
            const s3Image = wrapper.instance();

            await s3Image.load();

            expect.assertions(1);
            expect(spyon).toBeCalledWith('empty imgKey and path');

            spyon.mockClear();
        });

        test('only get image source when body no specified', async () => {
            const spyon = jest.spyOn(Storage, 'put').mockImplementationOnce(() => {
                return new Promise((res, rej) => {
                    res('text');
                });
            });

            const spyon2 = jest.spyOn(S3Image.prototype, 'getImageSource').mockImplementationOnce(() => {
                return;
            });

            const props = {
                imgKey: 'imgKey'
            };

            const wrapper = shallow(<S3Image/>);
            const s3Image = wrapper.instance();
            wrapper.setProps(props);

            await s3Image.load();

            expect.assertions(2);
            expect(spyon).not.toBeCalled();
            expect(spyon2).toBeCalled();

            spyon.mockClear();
            spyon2.mockClear();
        });
    });

    describe('handlePick test', () => {
        test('happy case', async () => {
            const spyon = jest.spyOn(Storage, 'put').mockImplementationOnce(() => {
                return new Promise((res, rej) => {
                    res('data');
                });
            });

            const spyon2 = jest.spyOn(S3Image.prototype, 'getImageSource').mockImplementationOnce(() => {
                return;
            });

            const wrapper = shallow(<S3Image/>);
            const s3Image = wrapper.instance();
            wrapper.setProps({
                imgKey: 'imgKey',
                level: 'level',
                fileToKey: 'fileToKey'
            });

            const data = {
                file: 'file',
                name: 'name',
                size: 'size',
                type: 'type'
            };

            await s3Image.handlePick(data);

            expect.assertions(2);
            expect(spyon).toBeCalledWith('imgKey', 'file', {"contentType": "type", "level": "level", "track": undefined});
            expect(spyon2).toBeCalled();

            spyon.mockClear();
            spyon2.mockClear();
        });

        test('storage put error', async () => {
            const spyon = jest.spyOn(Storage, 'put').mockImplementationOnce(() => {
                return new Promise((res, rej) => {
                    rej('err');
                });
            });

            const spyon2 = jest.spyOn(Logger.prototype, 'debug');

            const wrapper = shallow(<S3Image/>);
            const s3Image = wrapper.instance();
            wrapper.setProps({
                imgKey: 'imgKey',
                level: 'level',
                fileToKey: 'fileToKey'
            });
            
            const data = {
                file: 'file',
                name: 'name',
                size: 'size',
                type: 'type'
            };

            await s3Image.handlePick(data);

            spyon.mockClear();
            spyon2.mockClear();
        });
    });

    describe('handleClick test', () => {
        test('happy case', () => {
            const mockFn = jest.fn();
            const props = {
                onClick: mockFn
            }

            const wrapper = shallow(<S3Image/>);
            const s3Image = wrapper.instance();
            wrapper.setProps(props);

            s3Image.handleClick('evt');

            expect(mockFn).toBeCalledWith('evt');
        });

        test('if no onclick', () => {
            const mockFn = jest.fn();

            const wrapper = shallow(<S3Image/>);
            const s3Image = wrapper.instance();

            s3Image.handleClick('evt');
        });
    });

    describe('handleOnLoad test', () => {
        test('happy case', () => {
            const mockFn = jest.fn();
            const props = {
                onLoad: mockFn
            }
            const state = {
                src: 'src'
            }
            const wrapper = shallow(<S3Image/>);
            const s3Image = wrapper.instance();
            wrapper.setProps(props);
            wrapper.setState(state);

            s3Image.handleOnLoad('img');

            expect(mockFn).toBeCalledWith('src');
        });

        test('if no onLoad', () => {
            const mockFn = jest.fn();

            const wrapper = shallow(<S3Image/>);
            const s3Image = wrapper.instance();


            s3Image.handleOnLoad('evt');
        });
    });

    describe('handleOnError test', () => {
        test('happy case', () => {
            const mockFn = jest.fn();
            const props = {
                onError: mockFn
            }
            const state = {
                src: 'src'
            }
            const wrapper = shallow(<S3Image/>);
            const s3Image = wrapper.instance();
            wrapper.setProps(props);
            wrapper.setState(state);

            s3Image.handleOnError('err');

            expect(mockFn).toBeCalledWith('src');
        });
    });

    describe('ImageEl test', () => {
        test('null text', () => {
            const wrapper = shallow(<S3Image/>);
            const s3Image = wrapper.instance();

            expect(s3Image.imageEl(null, 'theme')).toBeNull();
        });
    });

    describe('componentDidmount test', () => {
        test('happy case', () => {
            const spyon = jest.spyOn(S3Image.prototype, 'load');

            const wrapper = mount(<S3Image/>);

            expect(spyon).toBeCalled();
            spyon.mockClear();
        });
    });

    describe('getImageSource test', () => {
        test('happy case', () => {
            const wrapper = shallow(<S3Image/>);
            const s3Image = wrapper.instance();

            const spyon = jest.spyOn(Storage, 'get').mockImplementationOnce(() => {
                return new Promise((res, rej) => {
                    res('url');
                });
            });

            s3Image.getImageSource('key', 'level', false);
            expect(spyon).toBeCalledWith('key', {level: 'level', track: false});
            spyon.mockClear();
        });

        test('error case', () => {
            const wrapper = shallow(<S3Image/>);
            const s3Image = wrapper.instance();

            const spyon = jest.spyOn(Storage, 'get').mockImplementationOnce(() => {
                return new Promise((res, rej) => {
                    rej('err');
                });
            });

            try {
                s3Image.getImageSource('key', 'level', false);
            } catch (e) {
                expect(e).not.toBeNull();
            }
    
            spyon.mockClear();
        });
    });
});
