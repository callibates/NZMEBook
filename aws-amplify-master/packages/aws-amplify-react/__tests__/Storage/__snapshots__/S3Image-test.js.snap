// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`S3Image render test render null if no test and no picker 1`] = `
<div
  style={Object {}}
>
  <div
    onClick={[Function]}
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <img
      onError={[Function]}
      onLoad={[Function]}
      src="data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw=="
    />
    <div />
  </div>
</div>
`;

exports[`S3Image render test render with picker true 1`] = `
<div
  style={Object {}}
>
  <div
    onClick={[Function]}
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <img
      onError={[Function]}
      onLoad={[Function]}
      src="data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw=="
    />
    <div />
  </div>
  <div>
    <PhotoPicker
      key="picker"
      onPick={[Function]}
      theme={
        Object {
          "a": Object {
            "color": "#007bff",
            "cursor": "pointer",
          },
          "actionRow": Object {
            "marginBottom": "15px",
          },
          "button": Object {
            "backgroundColor": "#fff",
            "backgroundImage": "none",
            "border": "1px solid transparent",
            "borderColor": "#ccc",
            "borderRadius": "4px",
            "color": "#333",
            "cursor": "pointer",
            "display": "inline-block",
            "fontSize": "14px",
            "fontWeight": 400,
            "lineHeight": "1.42857143",
            "marginBottom": "0",
            "padding": "6px 12px",
            "textAlign": "center",
            "touchAction": "manipulation",
            "userSelect": "none",
            "verticalAlign": "middle",
            "whiteSpace": "nowrap",
          },
          "col1": Object {
            "display": "inline-block",
            "width": "8.33333333%",
          },
          "col10": Object {
            "display": "inline-block",
            "width": "83.33333333%",
          },
          "col11": Object {
            "display": "inline-block",
            "width": "91.66666667%",
          },
          "col12": Object {
            "display": "inline-block",
            "width": "100%",
          },
          "col2": Object {
            "display": "inline-block",
            "width": "16.66666667%",
          },
          "col3": Object {
            "display": "inline-block",
            "width": "25%",
          },
          "col4": Object {
            "display": "inline-block",
            "width": "33.33333333%",
          },
          "col5": Object {
            "display": "inline-block",
            "width": "41.66666667%",
          },
          "col6": Object {
            "display": "inline-block",
            "width": "50%",
          },
          "col7": Object {
            "display": "inline-block",
            "width": "58.33333333%",
          },
          "col8": Object {
            "display": "inline-block",
            "width": "66.66666667%",
          },
          "col9": Object {
            "display": "inline-block",
            "width": "75%",
          },
          "container": Object {
            "color": "#212529",
            "fontFamily": "-apple-system,
                BlinkMacSystemFont,
                \\"Segoe UI\\",
                Roboto,
                \\"Helvetica Neue\\",
                Arial,
                sans-serif,
                \\"Apple Color Emoji\\",
                \\"Segoe UI Emoji\\",
                \\"Segoe UI Symbol\\"",
            "fontWeight": "400",
            "lineHeight": "1.5",
            "paddingLeft": "15px",
            "paddingRight": "15px",
            "textAlign": "left",
          },
          "errorSection": Object {
            "backgroundColor": "#f0ad4e",
            "border": "1px solid #eea236",
            "borderRadius": "4px",
            "color": "#fff",
            "marginBottom": "20px",
            "textAlign": "left",
          },
          "formContainer": Object {
            "textAlign": "center",
          },
          "formRow": Object {
            "marginBottom": "15px",
          },
          "formSection": Object {
            "backgroundColor": "#fff",
            "border": "1px solid #ddd",
            "borderRadius": "4px",
            "display": "inline-block",
            "marginBottom": "20px",
            "textAlign": "left",
            "width": "400px",
          },
          "hidden": Object {
            "display": "none",
          },
          "input": Object {
            "backgroundColor": "#fff",
            "backgroundImage": "none",
            "border": "1px solid #ccc",
            "borderRadius": "4px",
            "boxShadow": "inset 0 1px 1px rgba(0,0,0,.075)",
            "boxSizing": "border-box",
            "color": "#555",
            "display": "block",
            "fontSize": "14px",
            "height": "34px",
            "lineHeight": "1.42857143",
            "padding": "6px 12px",
            "transition": "border-color ease-in-out .15s,box-shadow ease-in-out .15s",
            "width": "100%",
          },
          "nav": Object {
            "margin": "7.5px",
          },
          "navBar": Object {
            "border": "1px solid transparent",
            "borderColor": "#e7e7e7",
            "position": "relative",
          },
          "navButton": Object {
            "backgroundColor": "#fff",
            "backgroundImage": "none",
            "border": "1px solid transparent",
            "borderColor": "#ccc",
            "borderRadius": "4px",
            "color": "#333",
            "cursor": "pointer",
            "display": "inline-block",
            "fontSize": "14px",
            "fontWeight": 400,
            "lineHeight": "1.42857143",
            "marginBottom": "8px",
            "marginTop": "8px",
            "padding": "6px 12px",
            "textAlign": "center",
            "touchAction": "manipulation",
            "userSelect": "none",
            "verticalAlign": "middle",
            "whiteSpace": "nowrap",
          },
          "navItem": Object {
            "display": "inline-block",
            "lineHeight": "20px",
            "padding": "10px 5px",
          },
          "navRight": Object {
            "textAlign": "right",
          },
          "pre": Object {
            "backgroundColor": "#f5f5f5",
            "border": "1px solid #ccc",
            "borderRadius": "4px",
            "color": "#333",
            "display": "block",
            "fontFamily": "Menlo,
                Monaco,
                Consolas,
                \\"Courier New\\",
                monospace",
            "fontSize": "13px",
            "lineHeight": "1.42857143",
            "margin": "0 0 10px",
            "overflow": "auto",
            "padding": "9.5px",
            "wordBreak": "break-all",
            "wordWrap": "break-word",
          },
          "sectionBody": Object {
            "padding": "15px",
          },
          "sectionFooter": Object {
            "backgroundColor": "#f5f5f5",
            "borderTop": "1px solid #ddd",
            "borderTopLeftRadius": "3px",
            "borderTopRightRadius": "3px",
            "color": "#333",
            "padding": "10px 15px",
          },
          "sectionHeader": Object {
            "backgroundColor": "#337ab7",
            "borderBottom": "1px solid transparent",
            "borderColor": "#337ab7",
            "borderTopLeftRadius": "3px",
            "borderTopRightRadius": "3px",
            "color": "#fff",
            "padding": "10px 15px",
            "textAlign": "center",
          },
          "signInButton": Object {
            "#facebook_signin_btn": Object {
              "backgroundColor": "#3b5998",
              "borderColor": "rgba(0,0,0,0.2)",
              "color": "#fff",
            },
            "#google_signin_btn": Object {
              "backgroundColor": "#dd4b39",
              "borderColor": "rgba(0,0,0,0.2)",
              "color": "#fff",
            },
            "display": "block",
            "fontSize": "14px",
            "marginTop": "2px",
            "overflow": "hidden",
            "padding": "6px 12px 6px 44px",
            "position": "relative",
            "textAlign": "left",
            "textOverflow": "ellipsis",
            "whiteSpace": "nowrap",
            "width": "100%",
          },
          "space": Object {
            "display": "inline-block",
            "width": "20px",
          },
        }
      }
    />
  </div>
</div>
`;

exports[`S3Image render test render with src exist 1`] = `
<div
  style={Object {}}
>
  <div
    onClick={[Function]}
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <img
      onError={[Function]}
      onLoad={[Function]}
      src="imageSrc"
    />
    <div />
  </div>
</div>
`;

exports[`S3Image render test render with translate text 1`] = `
<div
  style={Object {}}
>
  <div
    onClick={[Function]}
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <img
      onError={[Function]}
      onLoad={[Function]}
      src="translate"
    />
    <div />
  </div>
</div>
`;
